src/TriangleNodesProvisioner.cpp

```diff
@@ -120,11 +120,37 @@ namespace blitzdg {
         rycub =-xscub/Jcub;
         sycub = xrcub/Jcub;
 
+        // repair Jacobian components on nodal points as well
+        real_matrix_type& Jac = *J, D_r = *Dr, D_s = *Ds;
+
+        real_matrix_type& r_x = *rx;
+        real_matrix_type& r_y = *ry;
+        real_matrix_type& s_x = *sx;
+        real_matrix_type& s_y = *sy;
+
+        index_type Np = NumLocalPoints, K = NumElements;
+        real_matrix_type xr(Np, K), yr(Np, K), xs(Np, K), ys(Np, K);
+
+        xr = sum(D_r(ii,kk)*x(kk,jj), kk);
+        yr = sum(D_r(ii,kk)*y(kk,jj), kk);
+        xs = sum(D_s(ii,kk)*x(kk,jj), kk);
+        ys = sum(D_s(ii,kk)*y(kk,jj), kk);
+
+        Jac = xr*ys - xs*yr;
+
+        // Invert the 2x2 mapping matrix.
+        r_x = ys/Jac;
```

Comment:
Are you guaranteed that the Jacobian is nonzero here?

Reply:
It really ought to be. It's probably best to throw a run-time error somewhere earlier on if it is. I'll try to add a check somewhere. Generally non-positive means it ain't going to work.